utils/s2n_map.c

```diff
@@ -31,17 +31,20 @@
 
 #define S2N_INITIAL_TABLE_SIZE 1024
 
-static S2N_RESULT s2n_map_slot(struct s2n_map *map, struct s2n_blob *key, uint32_t *slot)
+static S2N_RESULT s2n_map_slot(const struct s2n_map *map, struct s2n_blob *key, uint32_t *slot)
 {
     union {
         uint8_t u8[32];
         uint32_t u32[8];
     } digest;
 
-    GUARD_AS_RESULT(s2n_hash_update(&map->sha256, key->data, key->size));
-    GUARD_AS_RESULT(s2n_hash_digest(&map->sha256, digest.u8, sizeof(digest)));
+    struct s2n_hash_state sha256;
 
-    GUARD_AS_RESULT(s2n_hash_reset(&map->sha256));
+    GUARD_AS_RESULT(s2n_hash_new(&sha256));
+    GUARD_AS_RESULT(s2n_hash_init(&sha256, S2N_HASH_SHA256));
+    GUARD_AS_RESULT(s2n_hash_update(&sha256, key->data, key->size));
+    GUARD_AS_RESULT(s2n_hash_digest(&sha256, digest.u8, sizeof(digest)));
```

Comment:
Is there a possible memory leak here if one of these lines fails (unlikely, I admit) and thus the method returns prior to calling `s2n_hash_free`?

Reply:
Probably best to use `DEFER_CLEANUP`

Reply:
updated to use DEFER_CLEANUP